drivers/timer/nrf_grtc_timer.c

```diff
@@ -383,8 +340,10 @@ int z_nrf_grtc_timer_capture_read(int32_t chan, uint64_t *captured_time)
 		 */
 		return -EBUSY;
 	}
-
-	capt_time = nrfy_grtc_sys_counter_cc_get(NRF_GRTC, chan);
+	result = nrfx_grtc_syscounter_cc_value_read(chan, &capt_time);
+	if (result != NRFX_SUCCESS) {
```

Comment:
Shouldn't this throw `-EPERM` only if `result == NRFX_ERROR_FORBIDDEN`?
Other cases could throw different errors.
Also, header needs to be updated with new error codes:
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/timer/nrf_grtc_timer.h#L173